### PR TITLE
fixes #74

### LINKED ISSUE #74: naming
rename:
- RestRserveRequest -> Request
- RestRserveResponse -> Response
- RestRserveApplication -> Application

What else?

### PR DESCRIPTION
**Pull request by dselivanov**

### CODE REVIEW

File: R/RestRserveAuthMiddleware.R [hunk 1]
```diff
@@ -4,13 +4,13 @@
...
-#' This class inherits [RestRserveMiddleware].
+#' Adds various authorizations to [Application].
+#' This class inherits [Middleware].
 #'
 #' @section Construction:
 #'
 #' ```
-#' RestRserveAuthMiddleware$new(auth_backend, routes, match = "exact",
+#' AuthMiddleware$new(auth_backend, routes, match = "exact",
 #'                              name = "AuthMiddleware")
```
Comment (artemklevtsov) [inline R/RestRserveAuthMiddleware.R]: Indention.

File: R/RestRserveAuthMiddleware.R [hunk 2]
```diff
@@ -29,13 +29,13 @@
...
-#' [RestRserveMiddleware] [RestRserveApplication]
+#' [Middleware] [Application]
 #'
 #' @family AuthBackend
 #'
-RestRserveAuthMiddleware = R6::R6Class(
-  "RestRserveAuthMiddleware",
-  inherit = RestRserveMiddleware,
+AuthMiddleware = R6::R6Class(
+  "AuthMiddleware",
```
Comment (artemklevtsov) [inline R/RestRserveAuthMiddleware.R]: Param name (`classname`).

File: R/RestRserveMiddleware.R [hunk 3]
```diff
@@ -47,7 +47,7 @@
 #'
 #' * `process_response` :: `funciotn`\cr
 #'   Function which takes 2 arguments - `request` and `response` objects (class
-#'   [RestRserveRequest] and [RestRserveResponse] correspondingly) and modify
+#'   [Request] and [Response] correspondingly) and modify
 #'   `request` and `response` or throw exception using `HTTPError` helper.
```
Comment (artemklevtsov) [inline R/RestRserveMiddleware.R]: May be link to `HTTPError`?

File: R/RestRserveResponse.R [hunk 4]
```diff
@@ -9,7 +9,7 @@
 #' @section Construction:
 #'
 #' ```
-#' RestRserveResponse$new(body = "", content_type = 'text/plain',
+#' Response$new(body = "", content_type = 'text/plain',
 #'                        headers = list_named(), status_code = 200L, encode = NULL)
```
Comment (artemklevtsov) [inline R/RestRserveResponse.R]: Indent.

File: inst/tinytest/test-app-content-handlers.R [hunk 5]
```diff
@@ -6,45 +6,45 @@ source("setup.R")
...
-request_404 = RestRserveRequest$new(path = "/404")
+request_404 = Request$new(path = "/404")
 rs = app$process_request(request_404)
 expect_equal(rs[[1]], '{"error":"404 Not Found"}')
 expect_equal(rs[[2]], "application/json")
 
 HTTPError$reset()
 #---------------
-request_post_json = RestRserveRequest$new(path = "/json",
+request_post_json = Request$new(path = "/json",
```
Comment (artemklevtsov) [inline inst/tinytest/test-app-content-handlers.R]: Indent/

File: inst/tinytest/test-app-content-handlers.R [hunk 6]
```diff
@@ -6,45 +6,45 @@ source("setup.R")
...
                                           method = "POST",
                                           body = charToRaw('{"hello" : "world"}'),
                                           content_type = 'application/json')
 rs = app$process_request(request_post_json)
 expect_equal(unserialize(rs[[1]]), list(hello = 'world'))
 
 #---------------
-request_post_json = RestRserveRequest$new(path = "/json",
+request_post_json = Request$new(path = "/json",
```
Comment (artemklevtsov) [inline inst/tinytest/test-app-content-handlers.R]: Indent/

File: inst/tinytest/test-cl-request.R [hunk 7]
```diff
@@ -126,7 +126,7 @@ params = list(
   "param2" = "value2"
 )
 b = make_multipart_body(params, files)
-r = RestRserveRequest$new(content_type = attr(b, 'content-type'))
+r = Request$new(content_type = attr(b, 'content-type'))
```
Comment (artemklevtsov) [inline inst/tinytest/test-cl-request.R]: Quotes. We prefer double.

File: inst/tinytest/test-cl-request.R [hunk 8]
```diff
@@ -162,14 +162,14 @@ r$headers[["accept"]] = "text/xml"
...
-r = RestRserveRequest$new()
+r = Request$new()
 expect_null(r$date)
 r$headers[["date"]] = "Sun, 04 Aug 2019 07:17:39 GMT"
 expect_true(inherits(r$date, "POSIXct"))
 expect_equal(as.numeric(r$date), 1564903059)
 
 
-r = RestRserveRequest$new(path = '/a',
+r = Request$new(path = '/a',
```
Comment (artemklevtsov) [inline inst/tinytest/test-cl-request.R]: Indent. Quotes.

File: inst/tinytest/test-cl-response.R [hunk 9]
```diff
@@ -130,22 +130,22 @@ status = 200L
...
+r = Response$new()
 expect_equal(r$status, "200 OK")
 r$set_status_code(400L)
 expect_equal(r$status, "400 Bad Request")
 
 
 
 # Test reset
-rs = RestRserveResponse$new(body = list(a = 'body'),
+rs = Response$new(body = list(a = 'body'),
```
Comment (artemklevtsov) [inline inst/tinytest/test-cl-response.R]: Indent. Quotes.

File: R/RestRserveAuthMiddleware.R (hunk 2 above)
Comment (dselivanov) [reply R/RestRserveAuthMiddleware.R]: ?

Comment (artemklevtsov) [reply R/RestRserveAuthMiddleware.R]: Should be `classname = "AuthMiddleware"`

**Pull request merged**